woocommerce/woocommerce-android
# [Dynamic Dashboard] Feedback card

Closes: #11407 


### Description
This PR adds logic to show the feedback card in the dashboard when it's needed.

There is still an ongoing discussion about the position of the card p1714639095329519/1714486421.996339-slack-C0354HSNUJH, I'll update it if needed on a separate PR.

### Testing instructions
1. Apply the following patch to force showing the feedback card
<details>
<summary>Patch</summary>

```patch
Index: WooCommerce/src/main/kotlin/com/woocommerce/android/FeedbackPrefs.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/WooCommerce/src/main/kotlin/com/woocommerce/android/FeedbackPrefs.kt b/WooCommerce/src/main/kotlin/com/woocommerce/android/FeedbackPrefs.kt
--- a/WooCommerce/src/main/kotlin/com/woocommerce/android/FeedbackPrefs.kt	(revision 342fc01f601b4ae1d67eeec1627f1d10c0472ce9)
+++ b/WooCommerce/src/main/kotlin/com/woocommerce/android/FeedbackPrefs.kt	(date 1714410744001)
@@ -20,6 +20,9 @@
 class FeedbackPrefs(
     private val context: Context
 ) {
+    init {
+        setString(LAST_FEEDBACK_DATE, Date(0).time.toString())
+    }
     private val gson by lazy { Gson() }
     private val featureMapTypeToken by lazy { object : TypeToken<Map<String, String>>() {}.type }
     private val sharedPreferences get() = PreferenceManager.getDefaultSharedPreferences(context)
@@ -32,7 +35,7 @@
             } ?: mapOf()
 
     val userFeedbackIsDue: Boolean
-        get() = AppPrefs.installationDate?.pastTimeDeltaFromNowInDays greaterThan THREE_MONTHS_IN_DAYS &&
+        get() = Date(0).pastTimeDeltaFromNowInDays greaterThan THREE_MONTHS_IN_DAYS &&
             lastFeedbackDate?.pastTimeDeltaFromNowInDays greaterThan SIX_MONTHS_IN_DAYS
 
     val userFeedbackIsDueObservable: Flow<Boolean> = callbackFlow {
```
</details>

2. Make sure the dynamic dashboard feature flag is enabled.
3. Open the app.
4. Confirm the feedback card is shown.
5. Click on one of the feedback card buttons.
6. Confirm the action works, and the card is hidden.
7. Close the app.
8. Repeat 3 to 7 with a different action (Positive, negative and dismiss), and confirm all work.

NB: The positive action won't do anything, as the debug app is not available in Google Play.

### Images/gif
<img width=360 src="https://github.com/woocommerce/woocommerce-android/assets/1657201/2d18372f-d6c2-456f-9ac6-96e85af9c94c" />
